src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -1683,20 +1683,40 @@ bool ParserNumber::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     char buf[MAX_LENGTH_OF_NUMBER + 1];
 
-    memcpy(buf, pos->begin, pos->size());
-    buf[pos->size()] = 0;
+    size_t size = pos->size();
+    memcpy(buf, pos->begin, size);
+    buf[size] = 0;
+    char * start_pos = buf;
 
-    char * pos_double = buf;
-    errno = 0;    /// Functions strto* don't clear errno.
-    Float64 float_value = std::strtod(buf, &pos_double);
-    if (pos_double != buf + pos->size() || errno == ERANGE)
+    if (*start_pos == '0')
     {
-        /// Try to parse number as binary literal representation. Example: 0b0001.
-        if (pos->size() > 2 && buf[0] == '0' && buf[1] == 'b')
+        ++start_pos;
+        --size;
+
+        /// binary
+        if (*start_pos == 'b')
         {
-            char * buf_skip_prefix = buf + 2;
+            ++start_pos;
+            --size;
```

Comment:
Looks like overflow is possible if  size=1 initially?

Reply:
Disregard, not possible due to second letter is already proven to be 'b', so size >=2